serialization/include/archives/SimdJsonInputArchive.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+
+#include <concepts>
+#include <cstdint>
+#include <exception>
+#include <fmt/format.h>
+#include <limits>
+#include <optional>
+#include <simdjson.h>
+#include <stdexcept>
+#include <string_view>
+#include <type_traits>
+
+#include "concepts/Concepts.h"
+
+namespace {
+
+struct SimdJsonSupportedTypeAdapter
+{
+  template <std::signed_integral T>
+  std::decay<int64_t> MapType(T);
```

Comment:
(no idea if you can talk to this bot, but they say you can, so let's try lol)

On purpose to have `ReturnType::type` return the target type , otherwise would get this
```
template <class T>
using SimdJsonSupportedType =
  typename decltype(std::declval<SimdJsonSupportedTypeAdapter>().MapType(
    std::declval<T>()));

Expected a qualified name after 'typename' clang(expected_qualified_after_typename)
```

If there's a cleaner way to do so, please suggest